File: client/src/app/datatypes/wordlist.ts
import { Word } from './word';

export class WordList {
  name: string;
  enabled: boolean;
  nouns: Word[];
  verbs: Word[];
  adjectives: Word[];
  misc: Word[];

  static getWordType(list: WordList,wordname: string ) {
    let result = '';
    [list.nouns, list.verbs, list.adjectives, list.misc].forEach(i => i.forEach(word => {
      if (word.word === wordname && result.length===0) {
        result = i === list.nouns ? 'Noun' : i === list.verbs ? 'Verb' : i === list.adjectives ? 'Adjective' : 'Misc';
      }
    }));
    return result;
  }
  static getWord(list: WordList,name: string ): Word {
    let result: Word;
    [list.nouns,list.verbs,list.adjectives,list.misc].forEach(i => i.forEach(word => {
      if(word.word === name) {result = word;}
    }));
    return result;
  }
}

File: server/src/main/java/umm3601/wordlists/WordList.java
package umm3601.wordlists;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class WordList {

  public String name;
  public boolean enabled;
  public Word[] nouns;
  public Word[] verbs;
  public Word[] adjectives;
  public Word[] misc;

  @JsonCreator
  public WordList(
    @JsonProperty("name") String name,
    @JsonProperty("enabled") boolean enabled,
    @JsonProperty("nouns") Word[] nouns,
    @JsonProperty("verbs") Word[] verbs,
    @JsonProperty("adjectives") Word[] adjectives,
    @JsonProperty("misc") Word[] misc
    ){
    this.name = name;
    this.enabled = enabled;
    this.nouns = nouns;
    this.verbs = verbs;
    this.adjectives = adjectives;
    this.misc = misc;
  }
}
